var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -169,3 +171,34 @@ def configure_args(self):
             args.append('--enable-burst-buffering')
 
         return args
+
+    examples_src_dir = 'examples/CXX'
+
+    @run_after('install')
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources([self.examples_src_dir])
+
+    def test(self):
+        test_dir = join_path(self.install_test_root, self.examples_src_dir)
+        with working_dir(test_dir, create=False):
```

Comment:
Performing the test using `with working_dir` is not needed when you're using `self.run_test`.  You should only need to supply the `work_dir` argument.

Reply:
@tldahlgren That's a good point. Perhaps `with working_dir` makes more sense when I actually need to create a temporary directory. I'll try setting only the `work_dir` argument, if it works I'll remove the `with working_dir`